spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala

```diff
@@ -396,34 +397,71 @@ object Sample {
       ArraySample(featureTensors)
     }
   }
+
+  def apply[T: ClassTag](
```

Comment:
The API looks not right. What's the meaning of T here?

Reply:
The base class Sample needs a type parameter T.

Reply:
I think we should change that. Or it's not a reasonable API.